docsource/architectureHighLevelDesign.rst

```diff
@@ -0,0 +1,27 @@
+Architecture - High Level Design
+=================================
+
+.. _fig-main:
+
+.. figure:: _static/kbaseSearchEngineDesignHighLevel.png
+
+    KBaseSearchEngine component diagram.
+
+.. figure:: _static/eventStateTransition.png
+    :scale: 50 %
+
+    Event state transition diagram.
+
+Data Flow
+---------
+1. The workspace pushes workspace level, object level and version level events into the KBaseSearchEngine Mongodb instance. The initial state of the events is UNPROC (or unprocessed).
+
+2. The EventQueue periodically fetches events from the database and sets those that can be processed into a READY state. The EventQueue is a three-level blocking queue that blocks events that may cause an out-of-order update on the index. For example, an object level event like "rename" must block another object level event like "delete". i.e. these two events cannot be executed in parallel by the workers.
```

Comment:
Also, importantly, it prevents simultaneous updates on the same document in ES, which can cause problems.

Reply:
Added note.